string_test.go

```diff
@@ -372,3 +372,138 @@ func TestAppend_KeyNotExist(t *testing.T) {
 	asserts.AssertIntReply(t, val, len(key))
 }
 
+func TestSetRange_StringExist(t *testing.T) {
+	testDB.Flush()
+	key := utils.RandString(10)
+	key2 := utils.RandString(3)
+	result := key2 + key[3:]
+	testDB.Exec(nil, utils.ToCmdLine2("SET", key, key))
+
+	actual := testDB.Exec(nil, utils.ToCmdLine("SetRange", key, fmt.Sprint(0), key2))
+	val, ok := actual.(*reply.BulkReply)
+	if !ok {
+		t.Errorf("expect nil bulk reply, get: %s", string(actual.ToBytes()))
+		return
+	}
+	asserts.AssertBulkReply(t, val, result)
+}
+
+func TestSetRange_StringExist_OffsetOutOfLen(t *testing.T) {
```

Comment:
I wonder what will happend if offset < len(bytes) but offset + len(value) > len(bytes). ^_^

Reply:
it will replace values in the end of value (from offset till end of value) and then append new values in the end, check 525 for...{} ^_^